### ISSUE TITLE
Cannot install valet, errors with symfony/process?

### ISSUE THREAD

**Issue by bradcis** [laravel/valet#489]
Hey guys,

I recently installed php 7.1.12 on my mac (high sierra), composer 1.5.5, laravel/installer, and now I'm trying to install laravel/valet to no avail. Below is the output I'm getting. Any ideas?

`composer global require laravel/valet
Changed current directory to /Users/brad/.composer
Using version ^2.0 for laravel/valet
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: don't install symfony/process v3.4.2
    - Conclusion: don't install laravel/valet v2.0.6
    - Conclusion: remove symfony/process v4.0.2
    - Conclusion: don't install symfony/process v3.4.1
    - Conclusion: don't install laravel/valet v2.0.5
    - Conclusion: don't install laravel/valet v2.0.4
    - Conclusion: don't install laravel/valet v2.0.5|don't install symfony/process v4.0.2|install symfony/process v2.7.0|install symfony/process v2.7.1|install symfony/process v2.7.10|install symfony/process v2.7.11|install symfony/process v2.7.12|install symfony/process v2.7.13|install symfony/process v2.7.14|install symfony/process v2.7.15|install symfony/process v2.7.16|install symfony/process v2.7.17|install symfony/process v2.7.18|install symfony/process v2.7.19|install symfony/process v2.7.2|install symfony/process v2.7.20|install symfony/process v2.7.21|install symfony/process v2.7.22|install symfony/process v2.7.23|install symfony/process v2.7.24|install symfony/process v2.7.25|install symfony/process v2.7.26|install symfony/process v2.7.27|install symfony/process v2.7.28|install symfony/process v2.7.29|install symfony/process v2.7.3|install symfony/process v2.7.30|install symfony/process v2.7.31|install symfony/process v2.7.32|install symfony/process v2.7.33|install symfony/process v2.7.34|install symfony/process v2.7.35|install symfony/process v2.7.36|install symfony/process v2.7.37|install symfony/process v2.7.38|install symfony/process v2.7.39|install symfony/process v2.7.4|install symfony/process v2.7.5|install symfony/process v2.7.6|install symfony/process v2.7.7|install symfony/process v2.7.8|install symfony/process v2.7.9|install symfony/process v2.8.0|install symfony/process v2.8.1|install symfony/process v2.8.10|install symfony/process v2.8.11|install symfony/process v2.8.12|install symfony/process v2.8.13|install symfony/process v2.8.14|install symfony/process v2.8.15|install symfony/process v2.8.16|install symfony/process v2.8.17|install symfony/process v2.8.18|install symfony/process v2.8.19|install symfony/process v2.8.2|install symfony/process v2.8.20|install symfony/process v2.8.21|install symfony/process v2.8.22|install symfony/process v2.8.23|install symfony/process v2.8.24|install symfony/process v2.8.25|install symfony/process v2.8.26|install symfony/process v2.8.27|install symfony/process v2.8.28|install symfony/process v2.8.29|install symfony/process v2.8.3|install symfony/process v2.8.30|install symfony/process v2.8.31|install symfony/process v2.8.32|install symfony/process v2.8.4|install symfony/process v2.8.5|install symfony/process v2.8.6|install symfony/process v2.8.7|install symfony/process v2.8.8|install symfony/process v2.8.9|install symfony/process v3.0.0|install symfony/process v3.0.1|install symfony/process v3.0.2|install symfony/process v3.0.3|install symfony/process v3.0.4|install symfony/process v3.0.5|install symfony/process v3.0.6|install symfony/process v3.0.7|install symfony/process v3.0.8|install symfony/process v3.0.9|install symfony/process v3.1.0|install symfony/process v3.1.1|install symfony/process v3.1.10|install symfony/process v3.1.2|install symfony/process v3.1.3|install symfony/process v3.1.4|install symfony/process v3.1.5|install symfony/process v3.1.6|install symfony/process v3.1.7|install symfony/process v3.1.8|install symfony/process v3.1.9|install symfony/process v3.2.0|install symfony/process v3.2.1|install symfony/process v3.2.10|install symfony/process v3.2.11|install symfony/process v3.2.12|install symfony/process v3.2.13|install symfony/process v3.2.14|install symfony/process v3.2.2|install symfony/process v3.2.3|install symfony/process v3.2.4|install symfony/process v3.2.5|install symfony/process v3.2.6|install symfony/process v3.2.7|install symfony/process v3.2.8|install symfony/process v3.2.9
    - Installation request for symfony/console (locked at v4.0.2) -> satisfiable by symfony/console[v4.0.2].
    - Conclusion: don't install symfony/process v4.0.2
    - laravel/valet v2.0.3 requires symfony/process ~2.7|~3.0 -> satisfiable by symfony/process[v2.7.0, v2.7.1, v2.7.10, v2.7.11, v2.7.12, v2.7.13, v2.7.14, v2.7.15, v2.7.16, v2.7.17, v2.7.18, v2.7.19, v2.7.2, v2.7.20, v2.7.21, v2.7.22, v2.7.23, v2.7.24, v2.7.25, v2.7.26, v2.7.27, v2.7.28, v2.7.29, v2.7.3, v2.7.30, v2.7.31, v2.7.32, v2.7.33, v2.7.34, v2.7.35, v2.7.36, v2.7.37, v2.7.38, v2.7.39, v2.7.4, v2.7.5, v2.7.6, v2.7.7, v2.7.8, v2.7.9, v2.8.0, v2.8.1, v2.8.10, v2.8.11, v2.8.12, v2.8.13, v2.8.14, v2.8.15, v2.8.16, v2.8.17, v2.8.18, v2.8.19, v2.8.2, v2.8.20, v2.8.21, v2.8.22, v2.8.23, v2.8.24, v2.8.25, v2.8.26, v2.8.27, v2.8.28, v2.8.29, v2.8.3, v2.8.30, v2.8.31, v2.8.32, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v2.8.8, v2.8.9, v3.0.0, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7, v3.0.8, v3.0.9, v3.1.0, v3.1.1, v3.1.10, v3.1.2, v3.1.3, v3.1.4, v3.1.5, v3.1.6, v3.1.7, v3.1.8, v3.1.9, v3.2.0, v3.2.1, v3.2.10, v3.2.11, v3.2.12, v3.2.13, v3.2.14, v3.2.2, v3.2.3, v3.2.4, v3.2.5, v3.2.6, v3.2.7, v3.2.8, v3.2.9, v3.3.0, v3.3.1, v3.3.10, v3.3.11, v3.3.12, v3.3.13, v3.3.14, v3.3.2, v3.3.3, v3.3.4, v3.3.5, v3.3.6, v3.3.7, v3.3.8, v3.3.9, v3.4.0, v3.4.1, v3.4.2].
    - laravel/valet v2.0.1 requires symfony/process ~2.7|~3.0 -> satisfiable by symfony/process[v2.7.0, v2.7.1, v2.7.10, v2.7.11, v2.7.12, v2.7.13, v2.7.14, v2.7.15, v2.7.16, v2.7.17, v2.7.18, v2.7.19, v2.7.2, v2.7.20, v2.7.21, v2.7.22, v2.7.23, v2.7.24, v2.7.25, v2.7.26, v2.7.27, v2.7.28, v2.7.29, v2.7.3, v2.7.30, v2.7.31, v2.7.32, v2.7.33, v2.7.34, v2.7.35, v2.7.36, v2.7.37, v2.7.38, v2.7.39, v2.7.4, v2.7.5, v2.7.6, v2.7.7, v2.7.8, v2.7.9, v2.8.0, v2.8.1, v2.8.10, v2.8.11, v2.8.12, v2.8.13, v2.8.14, v2.8.15, v2.8.16, v2.8.17, v2.8.18, v2.8.19, v2.8.2, v2.8.20, v2.8.21, v2.8.22, v2.8.23, v2.8.24, v2.8.25, v2.8.26, v2.8.27, v2.8.28, v2.8.29, v2.8.3, v2.8.30, v2.8.31, v2.8.32, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v2.8.8, v2.8.9, v3.0.0, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7, v3.0.8, v3.0.9, v3.1.0, v3.1.1, v3.1.10, v3.1.2, v3.1.3, v3.1.4, v3.1.5, v3.1.6, v3.1.7, v3.1.8, v3.1.9, v3.2.0, v3.2.1, v3.2.10, v3.2.11, v3.2.12, v3.2.13, v3.2.14, v3.2.2, v3.2.3, v3.2.4, v3.2.5, v3.2.6, v3.2.7, v3.2.8, v3.2.9, v3.3.0, v3.3.1, v3.3.10, v3.3.11, v3.3.12, v3.3.13, v3.3.14, v3.3.2, v3.3.3, v3.3.4, v3.3.5, v3.3.6, v3.3.7, v3.3.8, v3.3.9, v3.4.0, v3.4.1, v3.4.2].
    - laravel/valet v2.0.0 requires symfony/process ~2.7|~3.0 -> satisfiable by symfony/process[v2.7.0, v2.7.1, v2.7.10, v2.7.11, v2.7.12, v2.7.13, v2.7.14, v2.7.15, v2.7.16, v2.7.17, v2.7.18, v2.7.19, v2.7.2, v2.7.20, v2.7.21, v2.7.22, v2.7.23, v2.7.24, v2.7.25, v2.7.26, v2.7.27, v2.7.28, v2.7.29, v2.7.3, v2.7.30, v2.7.31, v2.7.32, v2.7.33, v2.7.34, v2.7.35, v2.7.36, v2.7.37, v2.7.38, v2.7.39, v2.7.4, v2.7.5, v2.7.6, v2.7.7, v2.7.8, v2.7.9, v2.8.0, v2.8.1, v2.8.10, v2.8.11, v2.8.12, v2.8.13, v2.8.14, v2.8.15, v2.8.16, v2.8.17, v2.8.18, v2.8.19, v2.8.2, v2.8.20, v2.8.21, v2.8.22, v2.8.23, v2.8.24, v2.8.25, v2.8.26, v2.8.27, v2.8.28, v2.8.29, v2.8.3, v2.8.30, v2.8.31, v2.8.32, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v2.8.8, v2.8.9, v3.0.0, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7, v3.0.8, v3.0.9, v3.1.0, v3.1.1, v3.1.10, v3.1.2, v3.1.3, v3.1.4, v3.1.5, v3.1.6, v3.1.7, v3.1.8, v3.1.9, v3.2.0, v3.2.1, v3.2.10, v3.2.11, v3.2.12, v3.2.13, v3.2.14, v3.2.2, v3.2.3, v3.2.4, v3.2.5, v3.2.6, v3.2.7, v3.2.8, v3.2.9, v3.3.0, v3.3.1, v3.3.10, v3.3.11, v3.3.12, v3.3.13, v3.3.14, v3.3.2, v3.3.3, v3.3.4, v3.3.5, v3.3.6, v3.3.7, v3.3.8, v3.3.9, v3.4.0, v3.4.1, v3.4.2].
    - Can only install one of: symfony/process[v3.0.0, v4.0.2].
    - Can only install one of: symfony/process[v3.0.1, v4.0.2].
    - Can only install one of: symfony/process[v3.0.2, v4.0.2].
    - Can only install one of: symfony/process[v3.0.3, v4.0.2].
    - Can only install one of: symfony/process[v3.0.4, v4.0.2].
    - Can only install one of: symfony/process[v3.0.5, v4.0.2].
    - Can only install one of: symfony/process[v3.0.6, v4.0.2].
    - Can only install one of: symfony/process[v3.0.7, v4.0.2].
    - Can only install one of: symfony/process[v3.0.8, v4.0.2].
    - Can only install one of: symfony/process[v3.0.9, v4.0.2].
    - Can only install one of: symfony/process[v3.1.0, v4.0.2].
    - Can only install one of: symfony/process[v3.1.1, v4.0.2].
    - Can only install one of: symfony/process[v3.1.10, v4.0.2].
    - Can only install one of: symfony/process[v3.1.2, v4.0.2].
    - Can only install one of: symfony/process[v3.1.3, v4.0.2].
    - Can only install one of: symfony/process[v3.1.4, v4.0.2].
    - Can only install one of: symfony/process[v3.1.5, v4.0.2].
    - Can only install one of: symfony/process[v3.1.6, v4.0.2].
    - Can only install one of: symfony/process[v3.1.7, v4.0.2].
    - Can only install one of: symfony/process[v3.1.8, v4.0.2].
    - Can only install one of: symfony/process[v3.1.9, v4.0.2].
    - Can only install one of: symfony/process[v3.2.0, v4.0.2].
    - Can only install one of: symfony/process[v3.2.1, v4.0.2].
    - Can only install one of: symfony/process[v3.2.10, v4.0.2].
    - Can only install one of: symfony/process[v3.2.11, v4.0.2].
    - Can only install one of: symfony/process[v3.2.12, v4.0.2].
    - Can only install one of: symfony/process[v3.2.13, v4.0.2].
    - Can only install one of: symfony/process[v3.2.14, v4.0.2].
    - Can only install one of: symfony/process[v3.2.2, v4.0.2].
    - Can only install one of: symfony/process[v3.2.3, v4.0.2].
    - Can only install one of: symfony/process[v3.2.4, v4.0.2].
    - Can only install one of: symfony/process[v3.2.5, v4.0.2].
    - Can only install one of: symfony/process[v3.2.6, v4.0.2].
    - Can only install one of: symfony/process[v3.2.7, v4.0.2].
    - Can only install one of: symfony/process[v3.2.8, v4.0.2].
    - Can only install one of: symfony/process[v3.2.9, v4.0.2].
    - Can only install one of: symfony/process[v3.3.0, v4.0.2].
    - Can only install one of: symfony/process[v3.3.1, v4.0.2].
    - Can only install one of: symfony/process[v3.3.10, v4.0.2].
    - Can only install one of: symfony/process[v3.3.11, v4.0.2].
    - Can only install one of: symfony/process[v3.3.12, v4.0.2].
    - Can only install one of: symfony/process[v3.3.13, v4.0.2].
    - Can only install one of: symfony/process[v3.3.14, v4.0.2].
    - Can only install one of: symfony/process[v3.3.2, v4.0.2].
    - Can only install one of: symfony/process[v3.3.3, v4.0.2].
    - Can only install one of: symfony/process[v3.3.4, v4.0.2].
    - Can only install one of: symfony/process[v3.3.5, v4.0.2].
    - Can only install one of: symfony/process[v3.3.6, v4.0.2].
    - Can only install one of: symfony/process[v3.3.7, v4.0.2].
    - Can only install one of: symfony/process[v3.3.8, v4.0.2].
    - Can only install one of: symfony/process[v3.3.9, v4.0.2].
    - Can only install one of: symfony/process[v3.4.0, v4.0.2].
    - Can only install one of: symfony/process[v2.7.0, v4.0.2].
    - Can only install one of: symfony/process[v2.7.1, v4.0.2].
    - Can only install one of: symfony/process[v2.7.10, v4.0.2].
    - Can only install one of: symfony/process[v2.7.11, v4.0.2].
    - Can only install one of: symfony/process[v2.7.12, v4.0.2].
    - Can only install one of: symfony/process[v2.7.13, v4.0.2].
    - Can only install one of: symfony/process[v2.7.14, v4.0.2].
    - Can only install one of: symfony/process[v2.7.15, v4.0.2].
    - Can only install one of: symfony/process[v2.7.16, v4.0.2].
    - Can only install one of: symfony/process[v2.7.17, v4.0.2].
    - Can only install one of: symfony/process[v2.7.18, v4.0.2].
    - Can only install one of: symfony/process[v2.7.19, v4.0.2].
    - Can only install one of: symfony/process[v2.7.2, v4.0.2].
    - Can only install one of: symfony/process[v2.7.20, v4.0.2].
    - Can only install one of: symfony/process[v2.7.21, v4.0.2].
    - Can only install one of: symfony/process[v2.7.22, v4.0.2].
    - Can only install one of: symfony/process[v2.7.23, v4.0.2].
    - Can only install one of: symfony/process[v2.7.24, v4.0.2].
    - Can only install one of: symfony/process[v2.7.25, v4.0.2].
    - Can only install one of: symfony/process[v2.7.26, v4.0.2].
    - Can only install one of: symfony/process[v2.7.27, v4.0.2].
    - Can only install one of: symfony/process[v2.7.28, v4.0.2].
    - Can only install one of: symfony/process[v2.7.29, v4.0.2].
    - Can only install one of: symfony/process[v2.7.3, v4.0.2].
    - Can only install one of: symfony/process[v2.7.30, v4.0.2].
    - Can only install one of: symfony/process[v2.7.31, v4.0.2].
    - Can only install one of: symfony/process[v2.7.32, v4.0.2].
    - Can only install one of: symfony/process[v2.7.33, v4.0.2].
    - Can only install one of: symfony/process[v2.7.34, v4.0.2].
    - Can only install one of: symfony/process[v2.7.35, v4.0.2].
    - Can only install one of: symfony/process[v2.7.36, v4.0.2].
    - Can only install one of: symfony/process[v2.7.37, v4.0.2].
    - Can only install one of: symfony/process[v2.7.38, v4.0.2].
    - Can only install one of: symfony/process[v2.7.39, v4.0.2].
    - Can only install one of: symfony/process[v2.7.4, v4.0.2].
    - Can only install one of: symfony/process[v2.7.5, v4.0.2].
    - Can only install one of: symfony/process[v2.7.6, v4.0.2].
    - Can only install one of: symfony/process[v2.7.7, v4.0.2].
    - Can only install one of: symfony/process[v2.7.8, v4.0.2].
    - Can only install one of: symfony/process[v2.7.9, v4.0.2].
    - Can only install one of: symfony/process[v2.8.0, v4.0.2].
    - Can only install one of: symfony/process[v2.8.1, v4.0.2].
    - Can only install one of: symfony/process[v2.8.10, v4.0.2].
    - Can only install one of: symfony/process[v2.8.11, v4.0.2].
    - Can only install one of: symfony/process[v2.8.12, v4.0.2].
    - Can only install one of: symfony/process[v2.8.13, v4.0.2].
    - Can only install one of: symfony/process[v2.8.14, v4.0.2].
    - Can only install one of: symfony/process[v2.8.15, v4.0.2].
    - Can only install one of: symfony/process[v2.8.16, v4.0.2].
    - Can only install one of: symfony/process[v2.8.17, v4.0.2].
    - Can only install one of: symfony/process[v2.8.18, v4.0.2].
    - Can only install one of: symfony/process[v2.8.19, v4.0.2].
    - Can only install one of: symfony/process[v2.8.2, v4.0.2].
    - Can only install one of: symfony/process[v2.8.20, v4.0.2].
    - Can only install one of: symfony/process[v2.8.21, v4.0.2].
    - Can only install one of: symfony/process[v2.8.22, v4.0.2].
    - Can only install one of: symfony/process[v2.8.23, v4.0.2].
    - Can only install one of: symfony/process[v2.8.24, v4.0.2].
    - Can only install one of: symfony/process[v2.8.25, v4.0.2].
    - Can only install one of: symfony/process[v2.8.26, v4.0.2].
    - Can only install one of: symfony/process[v2.8.27, v4.0.2].
    - Can only install one of: symfony/process[v2.8.28, v4.0.2].
    - Can only install one of: symfony/process[v2.8.29, v4.0.2].
    - Can only install one of: symfony/process[v2.8.3, v4.0.2].
    - Can only install one of: symfony/process[v2.8.30, v4.0.2].
    - Can only install one of: symfony/process[v2.8.31, v4.0.2].
    - Can only install one of: symfony/process[v2.8.32, v4.0.2].
    - Can only install one of: symfony/process[v2.8.4, v4.0.2].
    - Can only install one of: symfony/process[v2.8.5, v4.0.2].
    - Can only install one of: symfony/process[v2.8.6, v4.0.2].
    - Can only install one of: symfony/process[v2.8.7, v4.0.2].
    - Can only install one of: symfony/process[v2.8.8, v4.0.2].
    - Can only install one of: symfony/process[v2.8.9, v4.0.2].
    - Installation request for symfony/process (locked at v4.0.2) -> satisfiable by symfony/process[v4.0.2].
    - Installation request for laravel/valet ^2.0 -> satisfiable by laravel/valet[v2.0.0, v2.0.1, v2.0.3, v2.0.4, v2.0.5, v2.0.6].

Installation failed, reverting ./composer.json to its original content.`

**Comment by drbyte** [comment]
Seems like you've got other hard-coded dependencies in your `/Users/brad/.composer/composer.json`

Mind posting that file's contents?

**Comment by iberno** [comment]
Same Problem Yet!

**Comment by adamwathan** [comment]
Use CGR to avoid conflicts between global dependencies:

https://github.com/consolidation/cgr

**Comment by bradcis** [comment]
Thanks both! CGR fixed the issue :)

**Comment by almoncheng** [comment]
can you show me how to use CGR to fix the problem?

**Comment by adamwathan** [comment]
Should work exactly like the instructions say: https://github.com/consolidation/cgr#installation-and-usage

```
composer global require consolidation/cgr
cgr laravel/valet
valet install
```

**Comment by ChadTaljaardt** [comment]
I am getting the same error, is there no way to fix this without the work around?

**Comment by adamwathan** [comment]
Global composer packages share an autoloader so there is always potential
for annoying conflicts even though the two packages don’t need to work
together. You should use CGR even if you don’t use Valet.

**Comment by bloggervista** [comment]
no i just installed today and it doesn't work.

@adamwathan  i tried but it also don't work.

**Comment by drbyte** [comment]
Have you tried setting composer.json to use `dev-master` ? 
PR #498 was merged last week to add Symfony4 as a supported dependency.

**Comment by bloggervista** [comment]
@drbyte  i tried just using composer global require laravel/valet It shows error everywhere. I just did clean format on mac and now its not installing :(

**Comment by drbyte** [comment]
`composer global require laravel/valet:dev-master`

**Comment by bloggervista** [comment]
yea drbyte it works but i guess there is some problem with it

i got it installed anyway :)
```

  - Installing laravel/valet (dev-master 5df7ed7): Cloning 5df7ed7a1a
    Failed to download laravel/valet from source: Failed to clone https://github.com/laravel/valet.git, git was not found, check that it is installed and in your PATH env.

xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun

    Now trying to download from dist
  - Installing laravel/valet (dev-master 5df7ed7): Downloading (100%)
    Skipped installation of bin valet for package laravel/valet: name conflicts with an existing file
tightenco/collect suggests installing symfony/var-dumper (Required to prettify dd() and dump() (~3.3).)
Writing lock file
Generating autoload files
```

**Comment by amadeot** [comment]
same issue here, not working

**Comment by drbyte** [comment]
*sigh*  ... another "me too" with NO details.

1. WHICH "same issue" are *you* having?
2. What are the symptoms of YOUR "not working" scenario?
3. What's the content of your `~/.composer/composer.json` ?

**Comment by bloggervista** [comment]
laravel valet is not have installation problem but it caused many other to fail like laravel zero. I hope its fixed :)

**Comment by amadeot** [comment]
Sorry @drbyte ! I meant that I am/was having the exact same issue as the original issue. Wiped & reset a Macbook with High Sierra, used the same version of laravel/installer and composer, got the same error message. Ended up fixing it by using your suggestion of 

```
composer global require laravel/valet:dev-master
```

But forgot to come back and modify my comment. My bad!

**Comment by bloggervista** [comment]
@amadeot  that hack will fix problem but it causes problem for other package. so not recommended. I guess now normal install of laravel valet works

**Comment by drbyte** [comment]
@bloggervista you need to post more details about what EXACTLY are the problems you think dev-master is causing with "many others".  

We can't "fix it" when all you say is "I hope its fixed".  Seriously.

**Comment by bloggervista** [comment]
@drbyte  
4 day ago I tried to install laravel valet normally using 
    composer global require laravel/installer

But it wasn't installed at that time due to some error regarding dependencies like symfony/process.  I tried using cgr packaged mentioned somewhere etc.. but nothing worked

And i found this issue and installed dev version of valet using your command. And it worked like magic. And it was fine.

But after that i tried to install laravel-zero/installer globally but the error came. Tried everything to install that but there was some problem didn't thought it was valet. But after removing valet it installed fine. 

you can check at that issue

**Comment by drbyte** [comment]
Valet doesn't "need" symfony 4 components, but it doesn't reject installation if they are found. 
So, something else that was installed before valet may have pulled in the symfony 4 console component. Maybe because it's allowing "the most recent", and not prefer-oldest.

It's a matter of time before many other projects will update to be compatible with symfony 4 components. Hopefully other projects will investigate allowing compatibility of the newer components via composer.json rules if there isn't actually a breaking change.

**Comment by jonise92** [comment]
composer global require symfony/process:3.4
composer global require laravel/valet
valet install

**Comment by sagadsalem** [comment]
cgr command not found how to add it to my iterm

**Comment by bloggervista** [comment]
@sagadsalem `composer global require consolidation/cgr`

**Comment by thesogafoi** [comment]
You can run this command in your terminal

sudo apt-get install``php*-cli php*-common php*-curl php*-json php*-mbstring php*-mcrypt php*-opcache php*-readline php*-xml php*-zip

**Comment by OmarMakled** [comment]
Here is my solution 

- `mkdir valet && cd valet` // where all laravel projects located
- Download composer https://getcomposer.org/download/
-  Run `vendor/bin/valet install`

**Comment by misaelrpl** [comment]
Changed current directory to C:/Users/user/AppData/Roaming/Composer
./composer.json has been updated
Running composer update cretueusebiu/valet-windows
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - cretueusebiu/valet-windows[v0.1.0, ..., v0.1.2] require symfony/process ^3.0 -> found symfony/process[v3.0.0, ..., v3.4.47] but the package is fixed to v7.1.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - cretueusebiu/valet-windows[v0.1.3, ..., v0.1.8, v1.0.0, ..., v1.0.8, 2.0.3, ..., 2.0.14] require symfony/process ~2.7|~3.0 -> found symfony/process[v2.7.0, ..., v2.8.52, v3.0.0, ..., v3.4.47] but the package is fixed to v7.1.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - cretueusebiu/valet-windows[2.0.15, ..., 2.1.1] require symfony/process ~2.7|~3.0|~4.0 -> found symfony/process[v2.7.0, ..., v2.8.52, v3.0.0, ..., v3.4.47, v4.0.0, ..., v4.4.44] but the package is fixed to v7.1.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - cretueusebiu/valet-windows[2.1.2, ..., 2.1.5] require symfony/process ~2.7|~3.0|~4.0|~5.0 -> found symfony/process[v2.7.0, ..., v2.8.52, v3.0.0, ..., v3.4.47, v4.0.0, ..., v4.4.44, v5.0.0, ..., v5.4.40] but the package is fixed to v7.1.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - cretueusebiu/valet-windows[2.2.0, ..., 2.4.1] require symfony/process ^4.0|^5.0 -> found symfony/process[v4.0.0, ..., v4.4.44, v5.0.0, ..., v5.4.40] but the package is fixed to v7.1.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - cretueusebiu/valet-windows[2.5.0, ..., 2.5.2] require symfony/process ^4.0|^5.0|^6.0 -> found symfony/process[v4.0.0, ..., v4.4.44, v5.0.0, ..., v5.4.40, v6.0.0, ..., v6.4.8] but the package is fixed to v7.1.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires cretueusebiu/valet-windows * -> satisfiable by cretueusebiu/valet-windows[v0.1.0, ..., v0.1.8, v1.0.0, ..., v1.0.8, 2.0.3, ..., 2.5.2].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require cretueusebiu/valet-windows:*" to figure out if any version is installable, or "composer require cretueusebiu/valet-windows:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

I installed valet-windows and I got an error like this. Any ideas?

**Comment by drbyte** [comment]
@misaelrpl wrote:
> - cretueusebiu/valet-windows[2.5.0, ..., 2.5.2] require symfony/process ^4.0|^5.0|^6.0 -> found symfony/process[v4.0.0, ..., v4.4.44, v5.0.0, ..., v5.4.40, v6.0.0, ..., v6.4.8] but the package is fixed to v7.1.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

To be clear, this repository in which you posted this, is for `laravel/valet`, not `cretueusebiu/valet-windows`. It's normally best to post issues in the github repo for the project you're actually using.

As for that composer message, it means your composer setup is locked to (and therefore already has installed) Symfony 7, but your valet-windows project wants Symfony 6 specifically, and has not made itself compatible with newer Symfony versions.

Options:
- you could delete your `global` `composer.lock` file and maybe composer will be able to successfully downgrade its support for your other composer packages already installed, if compatible (if deleting `composer.lock`, it's best to also delete the `vendor` folder in the same directory, else the same error may result)
- you could review your global composer.json and see whether some packages you have there could be downgraded to an older version that's compatible with symfony/process v6 which your valet-windows wants
- You might consider https://herd.laravel.com for a more updated tool that has a supported Windows implementation.
